test/sqllogictest/role_membership.slt

```diff
@@ -46,6 +46,44 @@ SELECT * FROM role_members
 ----
 group1  joe  materialize
 
+# Dropping a role also removes it from role_members
+simple conn=mz_system,user=mz_system
+DROP ROLE group1
+----
+COMPLETE 0
+
+query TTT
+SELECT * FROM mz_role_members
+----
+
+statement ok
+CREATE ROLE group1
+
+statement ok
+GRANT group1 TO joe
+
+query TTT
+SELECT * FROM role_members
+----
+group1  joe  materialize
+
+# Dropping a granted role also removes it from role_members
+
+simple conn=mz_system,user=mz_system
+DROP ROLE joe
+----
+COMPLETE 0
+
+query TTT
+SELECT * FROM mz_role_members
```

Comment:
This currently fails because the table is not empty. If you do not want to fight this now and then again and again as you develop more tests, consider this:

- just create a separate .slt file for each distinct thing that is to be tested, rather than adding blocks to existing .slt files
- start each of the slt files with `reset-server` , which will ensure a pristine Mz instance that has no unexpected database objects flying around.

Reply:
No, this is a legitimate bug I believe: https://github.com/MaterializeInc/materialize/issues/18203